File: src/main/java/BB/resto/Entity/Commande.java
package BB.resto.Entity;

import BB.resto.Entity.Enumeration.EtatCommande;
import BB.resto.Entity.Enumeration.TypeRetrait;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Commande {
    private int id;
    private String numero;
    private int userId;
    private String adresse;
    private double montantTotal;
    private EtatCommande statut;
    private TypeRetrait typeRetrait;
    private Integer zoneId;
    private Integer livreurId;
    private String notes;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;


    private User client;
    private ZoneLivraison zoneLivraison;
    private User livreur;
    private List<LigneCommande> lignesCommande;


    public Commande() {
        this.statut = EtatCommande.EN_ATTENTE;
        this.createdAt = LocalDateTime.now();
        this.lignesCommande = new ArrayList<>();
    }

    public Commande(int userId, String adresse, TypeRetrait typeRetrait) {
        this();
        this.userId = userId;
        this.adresse = adresse;
        this.typeRetrait = typeRetrait;
    }


    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getNumero() { return numero; }
    public void setNumero(String numero) { this.numero = numero; }

    public int getUserId() { return userId; }
    public void setUserId(int userId) { this.userId = userId; }

    public String getAdresse() { return adresse; }
    public void setAdresse(String adresse) { this.adresse = adresse; }

    public double getMontantTotal() { return montantTotal; }
    public void setMontantTotal(double montantTotal) { this.montantTotal = montantTotal; }

    public EtatCommande getStatut() { return statut; }
    public void setStatut(EtatCommande statut) { this.statut = statut; }

    public TypeRetrait getTypeRetrait() { return typeRetrait; }
    public void setTypeRetrait(TypeRetrait typeRetrait) { this.typeRetrait = typeRetrait; }

    public Integer getZoneId() { return zoneId; }
    public void setZoneId(Integer zoneId) { this.zoneId = zoneId; }

    public Integer getLivreurId() { return livreurId; }
    public void setLivreurId(Integer livreurId) { this.livreurId = livreurId; }

    public String getNotes() { return notes; }
    public void setNotes(String notes) { this.notes = notes; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }


    public User getClient() { return client; }
    public void setClient(User client) { this.client = client; }

    public ZoneLivraison getZoneLivraison() { return zoneLivraison; }
    public void setZoneLivraison(ZoneLivraison zoneLivraison) { this.zoneLivraison = zoneLivraison; }

    public User getLivreur() { return livreur; }
    public void setLivreur(User livreur) { this.livreur = livreur; }

    public List<LigneCommande> getLignesCommande() { return lignesCommande; }
    public void setLignesCommande(List<LigneCommande> lignesCommande) { this.lignesCommande = lignesCommande; }


    public void addLigneCommande(LigneCommande ligne) {
        this.lignesCommande.add(ligne);
        ligne.setCommandeId(this.id);
    }

    public void calculerMontantTotal() {
        double total = 0;
        for (LigneCommande ligne : lignesCommande) {
            total += ligne.getSousTotal();
        }


        if (typeRetrait == TypeRetrait.LIVRAISON && zoneLivraison != null) {
            total += zoneLivraison.getPrixLivraison();
        }

        this.montantTotal = total;
    }

    public boolean estLivrable() {
        return typeRetrait == TypeRetrait.LIVRAISON && statut != EtatCommande.LIVREE
                && statut != EtatCommande.TERMINEE && statut != EtatCommande.ANNULEE;
    }

    public boolean peutEtreAnnulee() {
        return statut == EtatCommande.EN_ATTENTE || statut == EtatCommande.VALIDEE;
    }

    public String getFormattedMontant() {
        return String.format("%.2f FCFA", montantTotal);
    }

    @Override
    public String toString() {
        return "Commande{" +
                "id=" + id +
                ", numero='" + numero + '\'' +
                ", userId=" + userId +
                ", montantTotal=" + montantTotal +
                ", statut=" + statut +
                ", typeRetrait=" + typeRetrait +
                ", createdAt=" + createdAt +
                '}';
    }
}

File: BBurgerSymfony/BBurgerSymfony/src/Entity/Commande.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'commande')]
class Commande
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 30, unique: true)]
    private ?string $numero = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'user_id', nullable: false)]
    private ?User $user = null;

    #[ORM\Column(type: 'text')]
    private ?string $adresse = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?string $montantTotal = '0';

    #[ORM\Column(length: 20)]
    private ?string $statut = 'EN_ATTENTE';

    #[ORM\Column(length: 20)]
    private ?string $typeRetrait = null;

    #[ORM\ManyToOne(targetEntity: ZoneLivraison::class)]
    #[ORM\JoinColumn(name: 'zone_id', nullable: true)]
    private ?ZoneLivraison $zone = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'livreur_id', nullable: true)]
    private ?User $livreur = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $notes = null;

    #[ORM\Column(type: 'datetime')]
    private ?\DateTimeInterface $createdAt = null;

    #[ORM\Column(type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $updatedAt = null;

    #[ORM\OneToMany(targetEntity: LigneCommande::class, mappedBy: 'commande', cascade: ['persist', 'remove'])]
    private Collection $lignesCommande;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
        $this->lignesCommande = new ArrayCollection();
        $this->numero = $this->generateNumero();
    }

    private function generateNumero(): string
    {
        return 'CMD' . date('YmdHis') . rand(100, 999);
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNumero(): ?string
    {
        return $this->numero;
    }

    public function setNumero(string $numero): self
    {
        $this->numero = $numero;
        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;
        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): self
    {
        $this->adresse = $adresse;
        return $this;
    }

    public function getMontantTotal(): ?string
    {
        return $this->montantTotal;
    }

    public function setMontantTotal(string $montantTotal): self
    {
        $this->montantTotal = $montantTotal;
        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(string $statut): self
    {
        $this->statut = $statut;
        return $this;
    }

    public function getTypeRetrait(): ?string
    {
        return $this->typeRetrait;
    }

    public function setTypeRetrait(string $typeRetrait): self
    {
        $this->typeRetrait = $typeRetrait;
        return $this;
    }

    public function getZone(): ?ZoneLivraison
    {
        return $this->zone;
    }

    public function setZone(?ZoneLivraison $zone): self
    {
        $this->zone = $zone;
        return $this;
    }

    public function getLivreur(): ?User
    {
        return $this->livreur;
    }

    public function setLivreur(?User $livreur): self
    {
        $this->livreur = $livreur;
        return $this;
    }

    public function getNotes(): ?string
    {
        return $this->notes;
    }

    public function setNotes(?string $notes): self
    {
        $this->notes = $notes;
        return $this;
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function getUpdatedAt(): ?\DateTimeInterface
    {
        return $this->updatedAt;
    }

    public function setUpdatedAt(?\DateTimeInterface $updatedAt): self
    {
        $this->updatedAt = $updatedAt;
        return $this;
    }

    public function getLignesCommande(): Collection
    {
        return $this->lignesCommande;
    }

    public function addLigneCommande(LigneCommande $ligne): self
    {
        if (!$this->lignesCommande->contains($ligne)) {
            $this->lignesCommande->add($ligne);
            $ligne->setCommande($this);
        }
        return $this;
    }

    public function removeLigneCommande(LigneCommande $ligne): self
    {
        if ($this->lignesCommande->removeElement($ligne)) {
            if ($ligne->getCommande() === $this) {
                $ligne->setCommande(null);
            }
        }
        return $this;
    }

    public function getFormattedMontant(): string
    {
        return number_format((float)$this->montantTotal, 2, ',', ' ') . ' FCFA';
    }

    public function getStatutLabel(): string
    {
        return match($this->statut) {
            'EN_ATTENTE' => 'En attente',
            'VALIDEE' => 'Validée',
            'EN_PREPARATION' => 'En préparation',
            'PRETE' => 'Prête',
            'EN_LIVRAISON' => 'En livraison',
            'LIVREE' => 'Livrée',
            'TERMINEE' => 'Terminée',
            'ANNULEE' => 'Annulée',
            default => $this->statut
        };
    }
    
    
    public function peutEtreAnnulee(): bool
    {
        return in_array($this->statut, ['EN_ATTENTE', 'VALIDEE', 'EN_PREPARATION']);
    }
    
    
    public function peutEtreValidee(): bool
    {
        return $this->statut === 'EN_ATTENTE';
    }
    
    
    public function peutEtreMarqueePrete(): bool
    {
        return in_array($this->statut, ['VALIDEE', 'EN_PREPARATION']);
    }
    
    
    public function getNombreArticles(): int
    {
        $total = 0;
        foreach ($this->lignesCommande as $ligne) {
            $total += $ligne->getQuantite();
        }
        return $total;
    }
    

    public function getProduits(): array
    {
        $produits = [];
        foreach ($this->lignesCommande as $ligne) {
            $produits[] = [
                'produit' => $ligne->getProduit(),
                'quantite' => $ligne->getQuantite(),
                'prix_unitaire' => $ligne->getPrixUnitaire()
            ];
        }
        return $produits;
    }
}